flake.nix

```diff
@@ -0,0 +1,66 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+    flake-parts.url = "github:hercules-ci/flake-parts";
+    haskell-flake.url = "github:srid/haskell-flake";
+    pre-commit-hooks.url = "github:cachix/pre-commit-hooks.nix";
+  };
+  outputs = inputs@{ self, nixpkgs, flake-parts, ... }:
+    flake-parts.lib.mkFlake { inherit inputs; } {
+      systems = nixpkgs.lib.systems.flakeExposed;
+      imports = [
+        inputs.haskell-flake.flakeModule
+        inputs.pre-commit-hooks.flakeModule
+      ];
+
+      perSystem = { self', pkgs, ... }: {
+        haskellProjects.default = {
+          # The base package set representing a specific GHC version.
+          # By default, this is pkgs.haskellPackages.
+          # You may also create your own. See https://zero-to-flakes.com/haskell-flake/package-set
+          # basePackages = pkgs.haskellPackages;
+
+          # Extra package information. See https://zero-to-flakes.com/haskell-flake/dependency
+          #
+          # Note that local packages are automatically included in `packages`
+          # (defined by `defaults.packages` option).
+          #
+          # packages = {
+          #   aeson.source = "1.5.0.0"; # Hackage version override
+          #   shower.source = inputs.shower;
+          # };
+          # settings = {
+          #   aeson = {
+          #     check = false;
+          #   };
+          #   relude = {
+          #     haddock = false;
+          #     broken = false;
+          #   };
+          # };
+
+          devShell = {
+           # Enabled by default
+           enable = true;
+
+           # Programs you want to make available in the shell.
+           # Default programs can be disabled by setting to 'null'
+          #  tools = hp: { fourmolu = hp.fourmolu; ghcid = null; };
+
+           hlsCheck.enable = true;
+           pre-commit.settings.hooks = {
+            cabal-fmt.enable = true;
+            hlint.enable = true;
+            nixpkgs-fmt.enable = true;
+            ormolu.enable = true;
+          };
+          };
+
+
+        };
+
+        # haskell-flake doesn't set the default package, but you can do it here.
+        packages.default = self'.packages.amazonka-s3-streaming;
```

Comment:
I didn't seem to need to set this, contrary to what the comment implies.

Reply:
If I comment that out, `nix build` returns:
```
error: flake 'git+file:///Users/axman/Haskell/amazonka-s3-streaming' does not provide attribute 'packages.aarch64-darwin.default' or 'defaultPackage.aarch64-darwin'
```